src/Symfony/Component/Config/Definition/EnumNode.php

```diff
@@ -78,6 +84,10 @@ protected function finalizeValue(mixed $value): mixed
     {
         $value = parent::finalizeValue($value);
 
+        if ($this->enumFqcn && !$value instanceof \BackedEnum) {
+            $value = $this->enumFqcn::tryFrom($value);
+        }
```

Comment:
If the hydration fails, you're replacing the value given by the user with `null` although you might still need that value for producing a proper error message. I think, we'd be better of if we just used `::from()`, caught the `ValueException` and upcast it to an `InvalidConfigurationException`.